### PR TITLE
Adding option to map bitpay paid invoice status

### PR DESCRIPTION
**Pull request by farjad6**

### CODE REVIEW

File: BitPayLib/class-bitpayipnprocess.php [hunk 1]
```diff
@@ -262,8 +262,14 @@ private function process_refunded( Invoice $bitpay_invoice, WC_Order $order ): v
 
 	private function process_processing( Invoice $bitpay_invoice, WC_Order $order ): void {
 		$this->validate_bitpay_status_in_available_statuses( $bitpay_invoice, array( 'paid' ) );
-
 		$order->add_order_note( $this->get_start_order_note( $bitpay_invoice->getId() ) . 'is processing.' );
-		$order->update_status( 'processing', __( 'BitPay payment processing', 'woocommerce' ) );
+
+		$wordpress_order_status = $this->get_gateway_settings()['bitpay_checkout_order_process_paid_status'];
+		if ( WcGatewayBitpay::IGNORE_STATUS_VALUE === $wordpress_order_status ) {
```
Comment (shaunek-hero) [inline BitPayLib/class-bitpayipnprocess.php:268]: @farjad6 Over on the issue I created bobbrodie mentioned "best for a BitPay Paid Invoice Status setting, with a default matching today so we maintain backwards compatibility"... would this do that? Imagining a merchant who already had this plugin, then they get this update but they haven't yet gone into the settings page and updated... would it go to `wc-processing` status? It doesn't look like it to me, but just asking.

Comment (farjad6) [reply BitPayLib/class-bitpayipnprocess.php:268]: @shaunek-hero, yes sir. 
In previous code it was hardcoded to "processing". I am doing the same thing here: 
$new_status = $this->get_wc_order_statuses()[ $wordpress_order_status ] ?? 'processing';
That means if nothing is set take "processing".

**Comment by bobbrodie** [comment]
@farjad6 and @shaunek-hero thank you so much for the contribution! Prior to merging this in, could you update so that PHPCS passes? We are using [WordPress' PHP coding standards](https://github.com/WordPress/WordPress-Coding-Standards).

Example action is [here](https://github.com/bitpay/bitpay-checkout-for-woocommerce/actions/runs/7508308802/job/20471899341).

And this is the output, all minor stuff, all fixable by `phpcbf`:

```
FILE: ...pay-checkout-for-woocommerce/BitPayLib/class-wcgatewaybitpay.php
----------------------------------------------------------------------
FOUND 0 ERRORS AND 22 WARNINGS AFFECTING 22 LINES
----------------------------------------------------------------------
  59 | WARNING | [x] Array double arrow not aligned correctly;
     |         |     expected 35 space(s) between "'enabled'" and
     |         |     double arrow, but found 30.
  66 | WARNING | [x] Array double arrow not aligned correctly;
     |         |     expected 31 space(s) between "'bitpay_logo'" and
     |         |     double arrow, but found 26.
  87 | WARNING | [x] Array double arrow not aligned correctly;
     |         |     expected 19 space(s) between
     |         |     "'bitpay_logo_image_white'" and double arrow,
     |         |     but found 14.
  93 | WARNING | [x] Array double arrow not aligned correctly;
     |         |     expected 20 space(s) between
     |         |     "'bitpay_logo_image_dark'" and double arrow, but
     |         |     found 15.
  99 | WARNING | [x] Array double arrow not aligned correctly;
     |         |     expected 22 space(s) between
     |         |     "'bitpay_checkout_info'" and double arrow, but
     |         |     found 17.
 104 | WARNING | [x] Array double arrow not aligned correctly;
     |         |     expected 13 space(s) between
     |         |     "'bitpay_checkout_merchant_info'" and double
     |         |     arrow, but found 8.
 109 | WARNING | [x] Array double arrow not aligned correctly;
     |         |     expected 17 space(s) between
     |         |     "'bitpay_checkout_tier_info'" and double arrow,
     |         |     but found 12.
 114 | WARNING | [x] Array double arrow not aligned correctly;
     |         |     expected 31 space(s) between "'description'" and
     |         |     double arrow, but found 26.
 122 | WARNING | [x] Array double arrow not aligned correctly;
     |         |     expected 17 space(s) between
     |         |     "'bitpay_checkout_token_dev'" and double arrow,
     |         |     but found 12.
 130 | WARNING | [x] Array double arrow not aligned correctly;
     |         |     expected 16 space(s) between
     |         |     "'bitpay_checkout_token_prod'" and double arrow,
     |         |     but found 11.
 139 | WARNING | [x] Array double arrow not aligned correctly;
     |         |     expected 18 space(s) between
     |         |     "'bitpay_checkout_endpoint'" and double arrow,
     |         |     but found 13.
 150 | WARNING | [x] Array double arrow not aligned correctly;
     |         |     expected 22 space(s) between
     |         |     "'bitpay_checkout_flow'" and double arrow, but
     |         |     found 17.
 160 | WARNING | [x] Array double arrow not aligned correctly;
     |         |     expected 22 space(s) between
     |         |     "'bitpay_checkout_slug'" and double arrow, but
     |         |     found 17.
 165 | WARNING | [x] Array double arrow not aligned correctly;
     |         |     expected 20 space(s) between
     |         |     "'bitpay_custom_redirect'" and double arrow, but
     |         |     found 15.
 170 | WARNING | [x] Array double arrow not aligned correctly;
     |         |     expected 26 space(s) between
     |         |     "'bitpay_close_url'" and double arrow, but found
     |         |     21.
 175 | WARNING | [x] Array double arrow not aligned correctly;
     |         |     expected 22 space(s) between
     |         |     "'bitpay_checkout_mini'" and double arrow, but
     |         |     found 17.
 186 | WARNING | [x] Array double arrow not aligned correctly;
     |         |     expected 13 space(s) between
     |         |     "'bitpay_checkout_capture_email'" and double
     |         |     arrow, but found 8.
 197 | WARNING | [x] Array double arrow not aligned correctly;
     |         |     expected 10 space(s) between
     |         |     "'bitpay_checkout_checkout_message'" and double
     |         |     arrow, but found 5.
 203 | WARNING | [x] Array double arrow not aligned correctly;
     |         |     expected 21 space(s) between
     |         |     "'bitpay_checkout_error'" and double arrow, but
     |         |     found 16.
 209 | WARNING | [x] Array double arrow not aligned correctly;
     |         |     expected 13 space(s) between
     |         |     "'bitpay_checkout_error_message'" and double
     |         |     arrow, but found 8.
 253 | WARNING | [x] Array double arrow not aligned correctly;
     |         |     expected 6 space(s) between
     |         |     "'bitpay_checkout_order_expired_status'" and
     |         |     double arrow, but found 1.
 265 | WARNING | [x] Array double arrow not aligned correctly;
     |         |     expected [27](https://github.com/bitpay/bitpay-checkout-for-woocommerce/actions/runs/7508308802/job/20471899341#step:5:28) space(s) between "'bitpay_log_mode'"
     |         |     and double arrow, but found 22.
----------------------------------------------------------------------
PHPCBF CAN FIX THE 22 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```

Thanks again!

**Comment by farjad6** [comment]
> @farjad6 and @shaunek-hero thank you so much for the contribution! Prior to merging this in, could you update so that PHPCS passes? We are using [WordPress' PHP coding standards](https://github.com/WordPress/WordPress-Coding-Standards).
> 
> Example action is [here](https://github.com/bitpay/bitpay-checkout-for-woocommerce/actions/runs/7508308802/job/20471899341).
> 
> And this is the output, all minor stuff, all fixable by `phpcbf`:
> 
> ```
> FILE: ...pay-checkout-for-woocommerce/BitPayLib/class-wcgatewaybitpay.php
> ----------------------------------------------------------------------
> FOUND 0 ERRORS AND 22 WARNINGS AFFECTING 22 LINES
> ----------------------------------------------------------------------
>   59 | WARNING | [x] Array double arrow not aligned correctly;
>      |         |     expected 35 space(s) between "'enabled'" and
>      |         |     double arrow, but found 30.
>   66 | WARNING | [x] Array double arrow not aligned correctly;
>      |         |     expected 31 space(s) between "'bitpay_logo'" and
>      |         |     double arrow, but found 26.
>   87 | WARNING | [x] Array double arrow not aligned correctly;
>      |         |     expected 19 space(s) between
>      |         |     "'bitpay_logo_image_white'" and double arrow,
>      |         |     but found 14.
>   93 | WARNING | [x] Array double arrow not aligned correctly;
>      |         |     expected 20 space(s) between
>      |         |     "'bitpay_logo_image_dark'" and double arrow, but
>      |         |     found 15.
>   99 | WARNING | [x] Array double arrow not aligned correctly;
>      |         |     expected 22 space(s) between
>      |         |     "'bitpay_checkout_info'" and double arrow, but
>      |         |     found 17.
>  104 | WARNING | [x] Array double arrow not aligned correctly;
>      |         |     expected 13 space(s) between
>      |         |     "'bitpay_checkout_merchant_info'" and double
>      |         |     arrow, but found 8.
>  109 | WARNING | [x] Array double arrow not aligned correctly;
>      |         |     expected 17 space(s) between
>      |         |     "'bitpay_checkout_tier_info'" and double arrow,
>      |         |     but found 12.
>  114 | WARNING | [x] Array double arrow not aligned correctly;
>      |         |     expected 31 space(s) between "'description'" and
>      |         |     double arrow, but found 26.
>  122 | WARNING | [x] Array double arrow not aligned correctly;
>      |         |     expected 17 space(s) between
>      |         |     "'bitpay_checkout_token_dev'" and double arrow,
>      |         |     but found 12.
>  130 | WARNING | [x] Array double arrow not aligned correctly;
>      |         |     expected 16 space(s) between
>      |         |     "'bitpay_checkout_token_prod'" and double arrow,
>      |         |     but found 11.
>  139 | WARNING | [x] Array double arrow not aligned correctly;
>      |         |     expected 18 space(s) between
>      |         |     "'bitpay_checkout_endpoint'" and double arrow,
>      |         |     but found 13.
>  150 | WARNING | [x] Array double arrow not aligned correctly;
>      |         |     expected 22 space(s) between
>      |         |     "'bitpay_checkout_flow'" and double arrow, but
>      |         |     found 17.
>  160 | WARNING | [x] Array double arrow not aligned correctly;
>      |         |     expected 22 space(s) between
>      |         |     "'bitpay_checkout_slug'" and double arrow, but
>      |         |     found 17.
>  165 | WARNING | [x] Array double arrow not aligned correctly;
>      |         |     expected 20 space(s) between
>      |         |     "'bitpay_custom_redirect'" and double arrow, but
>      |         |     found 15.
>  170 | WARNING | [x] Array double arrow not aligned correctly;
>      |         |     expected 26 space(s) between
>      |         |     "'bitpay_close_url'" and double arrow, but found
>      |         |     21.
>  175 | WARNING | [x] Array double arrow not aligned correctly;
>      |         |     expected 22 space(s) between
>      |         |     "'bitpay_checkout_mini'" and double arrow, but
>      |         |     found 17.
>  186 | WARNING | [x] Array double arrow not aligned correctly;
>      |         |     expected 13 space(s) between
>      |         |     "'bitpay_checkout_capture_email'" and double
>      |         |     arrow, but found 8.
>  197 | WARNING | [x] Array double arrow not aligned correctly;
>      |         |     expected 10 space(s) between
>      |         |     "'bitpay_checkout_checkout_message'" and double
>      |         |     arrow, but found 5.
>  203 | WARNING | [x] Array double arrow not aligned correctly;
>      |         |     expected 21 space(s) between
>      |         |     "'bitpay_checkout_error'" and double arrow, but
>      |         |     found 16.
>  209 | WARNING | [x] Array double arrow not aligned correctly;
>      |         |     expected 13 space(s) between
>      |         |     "'bitpay_checkout_error_message'" and double
>      |         |     arrow, but found 8.
>  253 | WARNING | [x] Array double arrow not aligned correctly;
>      |         |     expected 6 space(s) between
>      |         |     "'bitpay_checkout_order_expired_status'" and
>      |         |     double arrow, but found 1.
>  265 | WARNING | [x] Array double arrow not aligned correctly;
>      |         |     expected [27](https://github.com/bitpay/bitpay-checkout-for-woocommerce/actions/runs/7508308802/job/20471899341#step:5:28) space(s) between "'bitpay_log_mode'"
>      |         |     and double arrow, but found 22.
> ----------------------------------------------------------------------
> PHPCBF CAN FIX THE 22 MARKED SNIFF VIOLATIONS AUTOMATICALLY
> ----------------------------------------------------------------------
> ```
> 
> Thanks again!

done

**Review by shaunek-hero — COMMENTED** [review]
Things look solid to me but I do have one small suggestion

**Review by bobbrodie — APPROVED** [review]

**Pull request merged**